### PR TITLE
docs: detect and fix broken links

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Validate internal links on CI.

#### What changes did you make? (Give an overview)

Same as https://github.com/eslint/eslint/pull/16837

#### Related Issues

https://github.com/eslint/eslint/issues/16833

#### Is there anything you'd like reviewers to focus on?
No

### CODE REVIEW

**Review by harish-sethuraman — APPROVED** [review]
LGTM, Will wait for one more review. Checked the pages where changes were made ✅ .

**Review by amareshsm — COMMENTED** [review]
When I tried `npm run lint:links` it is taking 3-4mins to complete. Is this fine?

**Review by amareshsm — COMMENTED** [review]
can we do this linting to staged files?

**Comment by nzakas** [comment]
Because this takes so long, I don't think we should run this as part of regular linting. Right now, `npm run lint` will run this and I think that could be confusing and frustrating if people don't know what's happening. We can definitely run this check in CI, but otherwise, I think we should just check precommit files locally and let people choose to run the link check only if they want.

**Comment by snitin315** [comment]
Makes sense. I will update the script to run on CI only.

**Comment by nzakas** [comment]
I think we should match this up: https://github.com/eslint/eslint/pull/16837#issuecomment-1432020186

File: .gitignore
Comment (snitin315) [inline .gitignore:112]:
```suggestion
```

File: tools/validate-links.js
Comment (mdjermanovic) [inline tools/validate-links.js:14]:
```suggestion
    "/docs"
```

Would this work the same? With `_site/docs`, I'm getting errors locally on Windows because filepaths have backslash separators.

```
  ✖ FAIL load _site\docs
  | operator: load
  | expected: 200 _site\docs
  |   actual: ENOENT: no such file or directory, open 'C:\milos\eslint.org\_site\docs'
  |       at: _site\index.html:193:22 <a href="/docs">...</a>
```

Comment (snitin315) [reply tools/validate-links.js:14]: Yes, it should work as this is present in`href="/docs"`. We can mention `href="/docs"` itself for more clarity. Are you facing errors for `_site/sponsors` too? If yes, we can use [`path.normalize()`](https://nodejs.org/api/path.html#pathnormalizepath) too.

Comment (mdjermanovic) [reply tools/validate-links.js:14]: You're right, this happens with `_site/sponsors` too.

Comment (snitin315) [reply tools/validate-links.js:14]: I'll push a fix

Comment (snitin315) [reply tools/validate-links.js]: @mdjermanovic I've pushed an update, I also tried it on a [windows runner on CI](https://github.com/eslint/eslint.org/actions/runs/4258455519/jobs/7409708999), seems to be working as expected. Can you also verify it once?

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Leaving this open for others to verify.

**Review by nzakas — APPROVED** [review]
LGTM. Nice work!

**Pull request merged**
